### PR TITLE
Fixes initialization flow and how timers are triggered

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/10510

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- defined in the issue

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/state/state_util.cc [hunk 1]
```diff
@@ -171,7 +171,13 @@ double GetAutoContributionAmount(bat_ledger::LedgerImpl* ledger) {
 
 uint64_t GetReconcileStamp(bat_ledger::LedgerImpl* ledger) {
   DCHECK(ledger);
-  return ledger->GetUint64State(ledger::kStateNextReconcileStamp);
+  auto stamp = ledger->GetUint64State(ledger::kStateNextReconcileStamp);
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/state/state_util.cc:174]: just as a safe guard we should never return 0 from this function

File: vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc [hunk 2]
```diff
@@ -99,28 +99,34 @@ LedgerImpl::~LedgerImpl() {
...
+  }
+
+  bat_publisher_->SetPublisherServerListTimer(GetRewardsMainEnabled());
+  bat_contribution_->SetReconcileTimer();
+  bat_promotion_->Refresh(false);
+  bat_contribution_->Initialize();
+  bat_promotion_->Initialize();
+  bat_api_->Initialize();
+  SetConfirmationsWalletInfo();
+  braveledger_recovery::Check(this);
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc:127]: This line was previously before `SetConfirmationsWalletInfo`. Was the reordering intentional?

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc:127]: not really, but it makes no difference

**Comment by NejcZdovc** [comment]
CI failed on `test-install`

**Comment by LaurenWags** [comment]
Verified using

```
Brave | 1.12.61 Chromium: 83.0.4103.116 (Official Build) nightly (64-bit)
-- | --
Revision | 8f0c18b4dca9b6699eb629be0f51810c24fb6428-refs/branch-heads/4103@{#716}
OS | macOS Version 10.14.6 (Build 18G3020)
```

* Verified STR from issue https://github.com/brave/brave-browser/issues/10510
* After enabling Rewards, I saw
```
[13556:775:0630/151429.840205:VERBOSE1:contribution.cc(264)] Timer will start in 180
```
in the logs, which is expected since I used `--rewards=staging=true,reconcile-interval=3` when launching.

* Confirmed after 3 min (180s) contribution timers started:
```
[13556:775:0630/151729.841770:VERBOSE1:contribution.cc(264)] Timer will start in 180
[13556:775:0630/151729.842283:VERBOSE1:contribution.cc(167)] Staring monthly contribution
[13556:775:0630/151729.844424:VERBOSE1:contribution.cc(264)] Timer will start in 2
[13556:775:0630/151729.845054:VERBOSE1:contribution_ac.cc(33)] Starting auto contribution
```

and my recurring tip was contributed as expected
Screen Shot 2020-06-30 at 3 18 11 PM

**Pull request merged**